### DISCUSSION TITLE
JavaScript Examples

### DISCUSSION THREAD

**Discussion by jarrodhroberson** [TanStack/query#4595] [Q&A]
This really needs some vanilla JavaScript examples.

If nothing else the "Simple" and "Basic" examples need to be in plain old JavaScript.

The Typescript obfuscates how to use the library to the point that it is hard to understand what it even does much less how you need to get anything done with it.

**Comment by TkDodo** [comment]
not sure what you are talking about at all. Most examples, including `simple` and `basic`, are written in plain JS. It's one of the points I want to change in the mid to long term...

https://tanstack.com/query/v4/docs/examples/react/simple
https://tanstack.com/query/v4/docs/examples/react/basic

**Comment by ghost** [comment]
https://www.w3schools.com/js/js_examples.asp
https://www.programiz.com/javascript/examples

Don't forget select as answer @jarrodhroberson

**Comment by JimSchofield** [comment]
I'm trying to simply run a mutation function in vanilla js.  I have the documentation for doing this using a hook:

https://tanstack.com/query/latest/docs/framework/react/guides/mutations

Using the query client I can query and cache the results easily.  How do I run a mutation in vanilla js?

**Reply by JimSchofield** [reply]
This might be the answer... But I think it's true that it is not clear how to use the vanilla js API to do certain things.

https://github.com/TanStack/query/discussions/4771#discussioncomment-5999962
